yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -475,6 +479,69 @@ export class Sequencer {
     return true;
   }
 
+  /**
+   * Build a block
+   *
+   * Shared between the sequencer and the validator for re-execution
+   *
+   * @param validTxs - The valid transactions to construct the block from
+   * @param newGlobalVariables - The global variables for the new block
+   * @param historicalHeader - The historical header of the parent
+   * @param interrupt - The interrupt callback, used to validate the block for submission and check if we should propose the block
+   */
+  private async buildBlock(
+    validTxs: Tx[],
+    newGlobalVariables: GlobalVariables,
+    historicalHeader?: Header,
+    interrupt?: (processedTxs: ProcessedTx[]) => Promise<void>,
+  ) {
+    this.log.debug('Requesting L1 to L2 messages from contract');
+    const l1ToL2Messages = await this.l1ToL2MessageSource.getL1ToL2Messages(newGlobalVariables.blockNumber.toBigInt());
+    this.log.verbose(
+      `Retrieved ${l1ToL2Messages.length} L1 to L2 messages for block ${newGlobalVariables.blockNumber.toNumber()}`,
+    );
+
+    const numRealTxs = validTxs.length;
+    const blockSize = Math.max(2, numRealTxs);
+
+    // Sync to the previous block at least
+    await this.worldState.syncImmediate(newGlobalVariables.blockNumber.toNumber() - 1);
+
+    // NB: separating the dbs because both should update the state
+    const publicProcessorFork = await this.worldState.fork();
```

Comment:
I'm slightly confused here, two forks, and the process processor seems to have one, but then be passed the other as part of building, if they are both updating along why does it need two? Is it to handle reverts after writes or? This is just me not knowing more than really something for this specific pr.

Reply:
The two forks were introduced here: https://github.com/AztecProtocol/aztec-packages/pull/9634/files#diff-1c2150b6e6b122e3a184b1132b8d6946e4e27fefe53381035810638fabcebe9c

It appears the txhandler in block builder must have a different db to the public processor